From 1be6dcaa4ee7ffaed693684b9d39651872040281 Mon Sep 17 00:00:00 2001
From: Rui Salvaterra <rsalvaterra@gmail.com>
Date: Fri, 19 Jun 2020 10:45:25 +0100
Subject: [PATCH] ARM Cortex-A9: build the userspace with Thumb-2 instructions

Thumb-2 code is denser than pure ARM, reducing RAM usage and improving
performance due to better instruction cache footprint.

There's no reason for not enabling this feature on other ARMv7 targets
(cortex-a7 and cortex-a8), but I don't have the hardware to test it.

Signed-off-by: Rui Salvaterra <rsalvaterra at gmail.com>
Tested-by: Tad <tad@spotco.us>
---
 include/target.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/target.mk b/include/target.mk
index b92cd7e467..8fbe3fedcb 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -234,6 +234,11 @@ ifeq ($(DUMP),1)
     CPU_TYPE = sparc
     CPU_CFLAGS_ultrasparc = -mcpu=ultrasparc
   endif
+  ifeq ($(ARCH),arm)
+    CPU_CFLAGS_cortex-a7 = -mthumb
+    CPU_CFLAGS_cortex-a8 = -mthumb
+    CPU_CFLAGS_cortex-a9 = -mthumb
+  endif
   ifeq ($(ARCH),aarch64)
     CPU_TYPE ?= generic
     CPU_CFLAGS_generic = -mcpu=generic
-- 
2.31.1

